firebase-test/src/documents-rules.test.ts

```diff
@@ -159,6 +159,140 @@ describe("Firestore security rules", () => {
+      const historyDoc = { id: "an-id", tree: "my-document", action: "/content/stuff",
+                           undoable: true, createdAt: mockTimestamp(),
+                           records: [], state: "complete"
+                         };
+      // remove specified props for validating the tests that require them
+      options?.remove?.forEach(prop => delete (historyDoc as any)[prop]);
+      // add additional props to test behavior of additional props
+      options?.add && Object.keys(options.add).forEach(prop => {
+        (historyDoc as any)[prop] = options.add?.[prop];
+      });
+      return historyDoc;
+    }
+    function specHistoryEntryParentDoc(options?: ISpecHisoryDoc) {
+      const historyDoc = {context_id: thisClass, network: noNetwork, teachers: [teacherId], uid: teacherId,
+        type: "problemDocument", key: "my-document", createdAt: mockTimestamp()};
+      // remove specified props for validating the tests that require them
+      options?.remove?.forEach(prop => delete (historyDoc as any)[prop]);
+      // add additional props to test behavior of additional props
+      options?.add && Object.keys(options.add).forEach(prop => {
+        (historyDoc as any)[prop] = options.add?.[prop];
+      });
+      return historyDoc;
+    }
+    
+    it("unauthed reads fail", async () => {
+      db = initFirestore();
+      await expectReadToFail(db, kDocumentHistoryDocPath);
+    });
+    
+    it("unauthed user cannot read. Parent doc does not exist", async () => {
+      db = initFirestore();
+      await expectReadToFail(db, kDocumentHistoryDocPath);
+    });
+    
+    it("unauthed user cannot read. Parent doc exists", async () => {
+      db = initFirestore();
+      await adminWriteDoc(kDocumentDocPath, specHistoryEntryParentDoc());
+      await expectReadToFail(db, kDocumentHistoryDocPath);
+    });
+    
+    it ("student can read their own history entries", async () => {
+      db = initFirestore(studentAuth);
+      await adminWriteDoc(kDocumentDocPath, specHistoryEntryParentDoc({add:{uid: studentId }}));
+      await adminWriteDoc(kDocumentHistoryDocPath, specHistoryEntryDoc());
+      await expectReadToSucceed(db, kDocumentHistoryDocPath);
+    });
+    
+    it ("student cannot read their own history entries if no parent", async () => {
+      db = initFirestore(studentAuth);
+      await adminWriteDoc(kDocumentHistoryDocPath, specHistoryEntryDoc());
+      await expectReadToFail(db, kDocumentHistoryDocPath);
+    });
+    
+    it ("teacher can read student history", async () => {
+      db = initFirestore(teacherAuth);
+      await adminWriteDoc(kDocumentDocPath, specHistoryEntryParentDoc({add:{uid: studentId }}));
+      await adminWriteDoc(kDocumentHistoryDocPath, specHistoryEntryDoc());
+      await expectReadToSucceed(db, kDocumentHistoryDocPath);
+    });
+    
+    it ("users authed from different portals cannot read each other's history entries", async () => {
```

Comment:
Do you think this test captures what we chatted about with respect to the portal entry points?

Reply:
It would be better if there was something like a `studentOtherPortalAuth` that included a different `platform_id`. However I see that none of the mock auths in `setup-rules-test` have a platform id. And I also see that none of the rules seem to be checking the platform_id (with the exception of the `preservesReadOnlySupportFields`).  So I'm pretty sure this test will fail. Ie. we aren't preventing users that have a matching user id from accessing data in other portals. 

So I guess I'd remove this test, and add a FIXME to the rules file about this. And/or you could add a test here that is marked as [failing](https://jestjs.io/docs/api#testfailingname-fn-timeout)

Because we are trying to get the code frozen today, probably it is best add the FIXME rather than work on the test.